lib/src/main/java/io/ably/lib/transport/WebSocketTransport.java

```diff
@@ -223,6 +223,7 @@ public void onClose(final int wsCode, final String wsReason, final boolean remot
 
         @Override
         public void onError(final Exception e) {
+            Log.e(TAG, "Unexpected exception ", e);
```

Comment:
This is presumably called if there is any error on the websocket connection, and as such it's not "Unexpected" - it's a routine event. I would have the error message say "Connection error" or something like that.

Reply:
Agree! But I was hoping, connection-related errors will be logged as a part of the exception message anyway. Also, I found some of the cases where it can throw IO related exceptions (https://github.com/TooTallNate/Java-WebSocket/blob/a76e0524ece10062ca98ffa74e20a9f9d8bbd51d/src/main/java/org/java_websocket/client/WebSocketClient.java#L964)
- But, connection-related errors might occur most of the time, so makes sense to keep it specific & relevant.